frontend/src/pages/EditProfile.js

```diff
@@ -13,50 +13,55 @@ const EditProfile = () => {
   const [twitter, setTwitter] = useState('');
   const [linkedin, setLinkedin] = useState('');
   const [errors, setErrors] = useState({});
-  const [bio, setBio] = useState('');  
-
+  const [bio, setBio] = useState('');
 
   useEffect(() => {
+
     const fetchUserData = async () => {
       try {
-        const response = await axios.get(`${process.env.REACT_APP_BACKEND_URL}/user/${username}`);
+        const response = await axios.get(`${process.env.REACT_APP_BACKEND_URL}/users/${username}`);
         const data = response.data;
         setAvatar(data.avatar);
         setEmail(data.email);
         setDiscord(data.discord);
         setTelegram(data.telegram);
         setTwitter(data.twitter || '');
         setLinkedin(data.linkedin || '');
-        setBio(data.bio || ''); 
+        setBio(data.bio || '');
 
       } catch (error) {
         console.error('Error fetching user data', error);
       }
     };
-  
-    fetchUserData();
-  }, [username]);
-  
-const handleAvatarChange = async (e) => {
-  const file = e.target.files[0];
-  if (file) {
-    const formData = new FormData();
-    formData.append('avatar', file);
-    formData.append('username', username); 
 
-    try {
-      const response = await axios.post(`${process.env.REACT_APP_BACKEND_URL}/upload-avatar`, formData, {
-        headers: {
-          'Content-Type': 'multipart/form-data',
-        },
-      });
+    const storedUsername = localStorage.getItem('username');
+    if (storedUsername && storedUsername !== username) {
+      navigate(`/edit-profile/${storedUsername}`);
+    }else {
+      fetchUserData();
+    }
+  }, [username, navigate]);
 
-      setAvatar(response.data.avatarUrl);
-    } catch (error) {
-      console.error('Error uploading avatar', error);
+  const handleAvatarChange = async (e) => {
+    const file = e.target.files[0];
+    if (file) {
+      const formData = new FormData();
+      formData.append('avatar', file);
+      formData.append('username', username);
+
+      try {
+        const response = await axios.post(`${process.env.REACT_APP_BACKEND_URL}/upload-avatar`, formData, {
+          headers: {
+            'Content-Type': 'multipart/form-data',
+          },
+        });
+
+        setAvatar(response.data.avatarUrl);
+      } catch (error) {
+        console.error('Error uploading avatar', error);
+      }
     }
-  }
-};
+  };
```

Comment:
Consider adding some validation for file type and size before uploading. This will prevent unnecessary server requests and potential attacks like `XSS`, `SSRF` etc.